syserol/regression.py

```diff
@@ -36,52 +36,40 @@ def function_elastic_net(function="ADCC"):
     df_variables = df_merged.drop(["subject"] + functions, axis=1)
 
     # perform regression
-    Y_pred = elasticNetFunc(df_variables, Y)
-
+    enet = elasticNetFunc(df_variables, Y)
+    Y_pred = cross_val_predict(enet, df_variables, Y, cv=len(Y), n_jobs=-1)
     return Y, Y_pred, np.sqrt(r2_score(Y, Y_pred))
 
 
 def function_prediction(tensorFac, function="ADCC", evaluation="all"):
     """ Predict functions using our decomposition and regression methods"""
     func, _ = importFunction()
+    indices = AlterIndices()
 
     Y = func[function]
     X = tensorFac[1][0]  # subjects x components matrix
-    dropped = np.nonzero(np.isnan(Y.to_numpy()))
-    X = X[np.isfinite(Y), :]
+    notAlter = np.delete(np.arange(181), indices)
+    dropped = np.unique(np.concatenate((np.nonzero(np.isnan(Y.to_numpy()))[0], notAlter)))
+    Y_notAlter = Y[dropped][np.isfinite(Y)]
+    X_notAlter = X[Y_notAlter.index]
+    Y_Alter = Y[indices][np.isfinite(Y)]
+    X_Alter = X[Y_Alter.index]
     Y = Y[np.isfinite(Y)]
+    X = X[Y.index]
+
+    enet = elasticNetFunc(X_Alter, Y_Alter)
 
-    Y_pred = elasticNetFunc(X, Y)
 
     if evaluation == "all":
+        Y_pred = cross_val_predict(enet, X, Y, cv=len(Y), n_jobs=-1)
```

Comment:
You only have to use cross validation if the observations you’re predicting are in the training data. Non-Alter subjects aren’t, so you can just call predict()

Reply:
Yes, this means you’ll have to separately handle the Alter and non-Alter subjects when collecting their predictions.